spec/integration/puppet_x/puppetlabs/scheduled_task/taskscheduler2_spec.rb

```diff
@@ -0,0 +1,177 @@
+
+    result = `#{full_cmd}`
+
+    result.strip
+  end
+
+  match do |actual|
+    from_ps = run_ps(ps_cmd)
+
+    # This matcher probably won't tolerate UTF8 characters
+    actual.to_s == from_ps
+  end
+
+  failure_message do |actual|
+    "expected that #{actual} would match #{run_ps(ps_cmd)} from PowerShell command #{ps_cmd}"
+  end
+end
+
+def create_test_task(task_name = nil, task_compatiblity = PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2::TASK_COMPATIBILITY_V2)
+  tasksched = PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2
+  task_name = tasksched::ROOT_FOLDER + 'puppet_task_' + SecureRandom.uuid.to_s if task_name.nil?
+  definition = tasksched.definition()
+  # Set task settings
+  tasksched.set_compatibility(definition, task_compatiblity)
+  tasksched.set_principal(definition, '')
+  definition.Settings.Enabled = false
+  # Create a trigger
+  trigger = tasksched.append_new_trigger(definition, tasksched::TASK_TRIGGER_TIME)
+  trigger.StartBoundary = '2017-09-11T14:02:00'
+  # Create an action
+  new_action = tasksched.create_action(definition, tasksched::TASK_ACTION_EXEC)
+  new_action.Path = 'cmd.exe'
+  new_action.Arguments = '/c exit 0'
+  tasksched.save(task_name, definition)
+
+  task_name
+end
+
+describe "PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2", :if => Puppet.features.microsoft_windows? do
+  let(:subject_taskname) { nil }
+  let(:subject) { PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2 }
+
+  describe '#enum_task_names' do
+    before(:all) do
+      # Need a V1 task as a test fixture
+      @task_name = create_test_task(nil, PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2::TASK_COMPATIBILITY_V1)
+    end
+
+    after(:all) do
+      PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2.delete(@task_name)
+    end
+
+    it 'should return all tasks by default' do
+      subject_count = subject.enum_task_names.count
+      ps_cmd = '(Get-ScheduledTask | Measure-Object).count'
+      expect(subject_count).to be_same_as_powershell_command(ps_cmd)
+    end
+
+    it 'should not recurse folders if specified' do
+      subject_count = subject.enum_task_names(PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2::ROOT_FOLDER, { :include_child_folders => false}).count
+      ps_cmd = '(Get-ScheduledTask | ? { $_.TaskPath -eq \'\\\' } | Measure-Object).count'
+      expect(subject_count).to be_same_as_powershell_command(ps_cmd)
+    end
+
+    it 'should only return compatible tasks if specified' do
+      subject_count = subject.enum_task_names(PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2::ROOT_FOLDER, { :include_compatibility => [PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2::TASK_COMPATIBILITY_V1]}).count
+      ps_cmd = '(Get-ScheduledTask | ? { [Int]$_.Settings.Compatibility -eq 1 } | Measure-Object).count'
+      expect(subject_count).to be_same_as_powershell_command(ps_cmd)
+    end
+  end
+
+  describe '#delete' do
+    before(:all) do
```

Comment:
I would probably change this from `:all` to `:each` in the event more tests are added to this `describe`

Reply:
Done